### DISCUSSION TITLE
Ender dragon

### DISCUSSION THREAD

**Discussion by deftesthawk** [sf-inc/sihywtcamd#23] [Ideas]
Maybe when the end dragon hits you with one of its fireball attacks it could respawn one of the end crystals that heal him.
Maybe each time you destroy one of the crystals he will spawn some enemies on you or it could enrage a few Endermen, this would stop people spamming all the crystals really fast and add more to the mechanic
Maybe you could add health to the end crystals so they are a bit stronger instead of being able to destroy them with 1 arrow, a health bar above each one would be good if you did this idea though

Some of these ideas are probably hard to implement

**Comment by charlyb01** [comment]
Fireballs respawning crystals... too broken I think. But maybe dragon breath could steal your life (and regen a bit the dragon). I'll think about it, what do you think?
I like the idea of end crystal desctruction that makes spawn ennemies (or enrage endermen) as the wither spawns wither skeletons when half health. That would make the dragon harder, even more than the wither maybe? That could make sense as it should be the "final boss".
For the health of end crystals, I'm not sure about this one. I'll think about it too.

**Reply by deftesthawk** [reply]
Sounds good, the end crystal part were mobs attack you was my favourite part of the idea. So if you end up doing something around that it will definitely add a much needed challenge to the encounter

**Comment by GalatynF** [comment]
Respawning crystals sounds really nice to make the fight last for longer, but it needs a better justification than "enderball hits you" (I don't see the logic in that).
What about : a certain number of crystals respawn every time the dragon's health dips below, for example, 2/3 and 1/3 of its total health ?

Not sure either about crystal health, but it reminds me of something I forgot when we first designed the mod : those fricking cages, man. They're just useless, they need to actually protect the crystal instead of just being defeated by one arrow passing through the hole

**Reply by deftesthawk** [reply]
That sounds a lot better, I was trying to think of ways it would respawn crystals when I came up with the idea I definitely prefer yours

**Reply by charlyb01** [reply]
Definitely yes

**Comment by elhertz** [comment]
The ender dragon is the final boss but her attacks are pretty lacking

**Here some attack ideas:**

**Gravity attack**: 
-- fast levitation followed by a drop
**Tail spin attack**: 
-- 360º spin attack deals damage and sends you flying (default attack sends you flying only if the dragon is looking at you afaik)
**Magic rays** (guardian rays): 
-- dragon shoots lethal magic rays at everyone so you are forced to take cover or DIE
**Magic aura attack** (the dragon aura is too strong for your mortal body): 
-- half heart damage per second ignoring protection enchantments for 20-30 seconds
**Homing fireballs**: 
-- is that even possible in minecraft?
**Berserker mode**: 
-- when flying and at low health it targets the first player that does damage when in berserker mode

**Dragon stats buff ideas:**
- Dragon gets 50% damage reduction from projectiles when at 50% health
- Life steal and resistance buff when on 10% health


Ran out of ideas. I'm on board of the respawning crystals too, sounds good.
I hope my ideas are not out of the scope of this mod. Regards.

**Reply by charlyb01** [reply]
Wow those are insane ideas! =D
I may add some attacks yeah, you're right about that, but not that much. Indeed, some are a bit out of the scope of this mod.
For homing fireballs, I can tell you that everything is possible, even in Minecraft... but yeah, isn't it too much?

**Reply by elhertz** [reply]
Yeah some are too much. I'm reading too much fantasy manga nowadays xd
Homing fireballs seems fun cause normal fireballs never hit you and you never pay attention to them imo
I'm glad you like some of my ideas. I'm looking forward to see some epic dragon fights in the future ☺️
